### PR TITLE
Add PhotonVisibilityExport module

### PR DESCRIPTION
**Pull request by dguff**
Introduce the PhotonVisibilityExport module, which evaluate the visibility of each OpDet in a grid of points inside the TPC. The PR also includes example fhicls and ROOT macros for building 2D and 3D visibility maps.

### CODE REVIEW

File: duneopdet/PhotonPropagation/PhotonVisibilityExport.fcl [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+{
+   module_type:		   "PhotonVisibilityExport"
+   photovis_label:	 	   "photovis" 
+   photovis_map:         "photovis"
+   voxel_dx:                "10.00" # voxel x mesh step (in cm)
+   voxel_dy:                "10.00" # voxel y mesh step (in cm)
+   voxel_dz:                "10.00" # voxel z mesh step (in cm)
+   vis_model:      "semianalytical" # visibility model to be used
+                                    # (pick one between "semianalytical", "compgraph")
+   do_refl:                  @local::dunefd_pdfastsim_par_ar_refl.DoReflectedLight
```
Comment (lpaulucc) [inline duneopdet/PhotonPropagation/PhotonVisibilityExport.fcl:15]: I am a bit puzzled here for VD: In principle there should be light reflected off from the anode but the following parameter is set to false...
Should there be two configurations, one for VD and another for HD here?

File: duneopdet/PhotonPropagation/PhotonVisibilityExport.fcl [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+   do_include_anode_refl:    false
+   vuvhitspars:              @local::dunefd_pdfastsim_par_ar.VUVHits
+   vishitspars:              @local::dunefd_pdfastsim_par_ar_refl.VISHits
+   tfloaderpars:			       @local::dunevd_pdfastsim_ann_ar.TFLoaderTool
+
+   do_include_buffer:       false
```
Comment (lpaulucc) [inline duneopdet/PhotonPropagation/PhotonVisibilityExport.fcl:21]: Is it possible to have this parameter true? If not, do you mind adding a comment here?

**Review by lpaulucc — COMMENTED** [review]
Thank you so much for this, Daniele. I have a few comments, it might be a good idea to have specific configurations for HD/VD, Ar/Xe in the module's fhicl. What do you think?

**Comment by aolivier23** [comment]
@dguff is this still a work in progress, or could it be ready to merge in 3-4 hours?  I'd be happy to get it in to this week's release if you can answer the outstanding question and fix some of the conflicts with develop.

**Review by lpaulucc — APPROVED** [review]
Thank you for the changes, Daniele!

**Comment by dguff** [comment]
Hi @lpaulucc @aolivier23, thank you for your comments! I still have to check that the changes I have made do actually work, I will try to get back to you as soon as I can and hopefully before 3-4 hours

File: duneopdet/PhotonPropagation/PhotonVisibilityExport.fcl (hunk 1 above)
Comment (dguff) [reply duneopdet/PhotonPropagation/PhotonVisibilityExport.fcl:15]: Hi @lpaulucc, I have created three separate configurations for FD HD/VD-Ar/VD-Xe handling the reflection options like this
```
...
photovisexport_fdvd_ar.do_refl: @local::dunevd_pdfastsim_par_ar.DoReflectedLight
photovisexport_fdvd_ar.do_include_anode_refl: @local::dunevd_pdfastsim_par_ar.IncludeAnodeReflections
...
photovisexport_fdvd_xe.do_refl: @local::dunevd_pdfastsim_par_xe.DoReflectedLight
photovisexport_fdvd_xe.do_include_anode_refl: @local::dunevd_pdfastsim_par_xe.IncludeAnodeReflections
```
However if I check what is going on with a fhicl dump I still get no reflection for Xe light (at least for v09_88_00d00). Am I using the wrong reference?

**Comment by lpaulucc** [comment]
Yeah, I have just made a fhicl-dump of the VD config and indeed you find in there
         DoReflectedLight: false
         IncludeAnodeReflections: false
for both Ar and Xe. I think now I remember Jiaoyang saying she didn't find it necessary to have the anode reflection separated from the rest. So my bad! Sorry for that!

**Comment by dguff** [comment]
Unfortunately I could not make it @aolivier23, this one has to wait for the next release :crying_cat_face:

**Pull request merged**